x/wasm/types/tx.go

```diff
@@ -163,6 +174,35 @@ func (msg MsgExecuteContract) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{sdk.WasmToAccAddress(senderAddr)}
 }
 
+func (msg MsgExecuteContract) FnSignatureInfo() (string, int, error) {
+	if err := msg.Msg.ValidateBasic(); err != nil {
+		return "", 0, fmt.Errorf("failed to validate msg:%v", err)
+	}
+
+	var v interface{}
+	json.Unmarshal(msg.Msg, &v)
+	data := v.(map[string]interface{})
+	if len(data) != 1 {
+		return "", 0, fmt.Errorf("failed to check msg method:%s", string(msg.Msg.Bytes()))
+	}
+
+	method := ""
+	for k, _ := range data {
+		method = k
+		break
+	}
+
+	if len(method) <= 0 {
```

Comment:
```suggestion
	if len(method) == 0 {
```

Reply:
I's can be compatibility and robustness.